os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -965,8 +1099,55 @@ static void amebasmart_spi_exchange_nodma(FAR struct spi_dev_s *dev,
 			}
 		}
 	}
+#else
+/*************************************************************************
+ * Name: spi_exchange (with DMA)
+ *
+ * Description:
+ *   Exchange a block of data on SPI using DMA
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   txbuffer - A pointer to the buffer of data to be sent
+ *   rxbuffer - A pointer to a buffer in which to receive data
+ *   nwords   - the length of data to be exchanged in units of words.
+ *              The wordsize is determined by the number of bits-per-word
+ *              selected for the SPI interface.  If nbits <= 8, the data is
+ *              packed into uint8_t's; if nbits >8, the data is packed into uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+
+	int mode_16bit = 1;
+	spi_hookmastercallback(priv);
+	if (amebasmart_spi_9to16bitmode(priv)) {
+		/* 16-bit mode */
+		mode_16bit++;
+	}
+
+	if (txbuffer && rxbuffer) {
+		if (nwords * mode_16bit > 4096) PANIC();
```

Comment:
Dear @Taejun-Kwon, this is to sync up with abhishek's testing branch to avoid any confusion. Once verification is all done, i will discuss again to see how to revise this part.